clients/src/main/java/org/apache/kafka/common/record/LazyDownConversionRecordsSend.java

```diff
@@ -49,39 +49,28 @@ public LazyDownConversionRecordsSend(String destination, LazyDownConversionRecor
     public long writeTo(GatheringByteChannel channel, long previouslyWritten, int remaining) throws IOException {
         if (convertedRecordsWriter == null || convertedRecordsWriter.completed()) {
             MemoryRecords convertedRecords;
-
             // Check if we have more chunks left to down-convert
             if (convertedRecordsIterator.hasNext()) {
                 // Get next chunk of down-converted messages
                 ConvertedRecords<MemoryRecords> recordsAndStats = convertedRecordsIterator.next();
                 convertedRecords = recordsAndStats.records();
-
-                int sizeOfFirstConvertedBatch = convertedRecords.batchIterator().next().sizeInBytes();
-                if (previouslyWritten == 0 && sizeOfFirstConvertedBatch > size())
-                    throw new EOFException("Unable to send first batch completely." +
-                            " maximum_size: " + size() +
-                            " converted_records_size: " + sizeOfFirstConvertedBatch);
-
                 recordConversionStats.add(recordsAndStats.recordConversionStats());
-                log.debug("Got lazy converted records for partition {} with length={}", topicPartition(), convertedRecords.sizeInBytes());
+                log.debug("Got converted records for partition {} with length={}", topicPartition(), convertedRecords.sizeInBytes());
```

Comment:
nit: "Got" is not very descriptive. Maybe the message could be "Down-converted records partition ..."?